src/core/dbus-manager.c

```diff
@@ -1873,17 +1874,35 @@ static int method_switch_root(sd_bus_message *message, void *userdata, sd_bus_er
         }
 
         /* Safety check */
-        if (isempty(init)) {
-                r = path_is_os_tree(root);
+        r = path_is_os_tree(root);
+        if (r < 0)
+                return sd_bus_error_set_errnof(error, r,
+                                               "Failed to determine whether root path '%s' contains an OS tree: %m",
+                                               root);
+        if (r == 0)
+                return sd_bus_error_setf(error, SD_BUS_ERROR_INVALID_ARGS,
+                                         "Specified switch root path '%s' does not seem to be an OS tree. os-release file is missing.",
+                                         root);
+
+        if (!in_initrd()) {
```

Comment:
I may misunderstand the point of the commit, but now we allow to switch root from host -> initrd ??

Reply:
I once thought it was a valid case, but yeah we don't (and shouldn't) do exitrd transition through `systemctl` or `SwitchRoot()`... Will rework.